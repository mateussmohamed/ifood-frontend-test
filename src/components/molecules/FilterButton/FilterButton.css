.filter-button {
  min-width: 80px;
  border-radius: 20px;
  padding: 6px 10px;
  text-align: center;
  background-color: #0e1537;
  color: #ffffff;
  border: 1px solid #ffffff;
  margin-right: 10px;
  transition: all 0.5s;
  font-size: 14px;
  font-weight: 300;
}

.filter-button:hover,
.filter-button.active {
  cursor: pointer;
  background: #ffffff;
  color: #0e1537;
}

.filter-button:hover,
.filter-button:active,
.filter-button:focus {
  outline: 0;
}
