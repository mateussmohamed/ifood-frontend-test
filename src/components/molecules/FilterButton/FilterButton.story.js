import { storiesOf } from '@storybook/react';
import React from 'react';
import FilterButton from './index';

const stories = storiesOf('FilterButton', module);

stories.add('whit text prop', () => (
  <div style={{ float: 'left' }}>
    <FilterButton text="Brasil" />
    <FilterButton text="México" />
    <FilterButton text="Argentina" />
    <FilterButton text="Colômbia" />
  </div>
));
