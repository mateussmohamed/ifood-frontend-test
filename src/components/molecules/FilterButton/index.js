import React from 'react';
import PropTypes from 'prop-types';
import './FilterButton.css';

import FilterTitle from '../../atoms/FilterTitle';

const FilterButton = ({ text, action, active }) => (
  <button className={`filter-button ${active ? 'active' : ''}`} onClick={action}>
    <FilterTitle text={text} />
  </button>
);

FilterButton.propTypes = {
  active: PropTypes.bool,
  text: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default FilterButton;
