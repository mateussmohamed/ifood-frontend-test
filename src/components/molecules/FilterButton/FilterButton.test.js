import React from 'react';
import renderer from 'react-test-renderer';
import FilterButton from './index';

describe('FilterButton', () => {
  test('should FilterButton match to snapshot', () => {
    const component = (
      <FilterButton text="awesome filter button" action={e => e} />
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
