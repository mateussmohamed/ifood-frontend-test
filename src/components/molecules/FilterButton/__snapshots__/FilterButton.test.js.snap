// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton should FilterButton match to snapshot 1`] = `
<button
  className="filter-button"
  onClick={[Function]}
>
  <div
    className="filter-title"
  >
    awesome filter button
  </div>
</button>
`;
